dokka-subprojects/analysis-kotlin-api/src/test/kotlin/org/jetbrains/dokka/analysis/test/sample/SampleRewriterTest.kt

```diff
@@ -0,0 +1,491 @@
+                                assertFailsWith<IndexOutOfBoundsException> { list.elementAt(3) }
+                    
+                                val emptyList = emptyList<Int>()
+                                assertFailsWith<IndexOutOfBoundsException> { emptyList.elementAt(0) }
+                            }
+                        }
+                    }
+                """
+            }
+        }
+
+        testProject.useServices { context ->
+            val sample = sampleAnalysisEnvironmentCreator.use {
+                resolveSample(
+                    sourceSet = context.singleSourceSet(),
+                    fullyQualifiedLink = "samples.collections.Collections.Elements.elementAt"
+                )
+            }
+            assertNotNull(sample)
+
+            val expectedImports = listOf(
+                "kotlin.test.*"
+            )
+
+            val expectedBody = """
+                    val list = listOf(1, 2, 3)
+                    println(list.elementAt(0)) // 1
+                    println(list.elementAt(2)) // 3
+                    // list.elementAt(3) // will fail with IndexOutOfBoundsException
+                    
+                    val emptyList = emptyList<Int>()
+                    // emptyList.elementAt(0) // will fail with IndexOutOfBoundsException
+            """.trimIndent()
+
+            assertEquals(expectedImports, sample.imports)
+            assertEquals(expectedBody, sample.body)
+        }
+    }
+
+    @Test
+    fun `should rewrite call of assertFalse with fully qualified name`() {
+        val testProject = kotlinJvmTestProject {
+            plugin(SomeSampleRewriterPlugin())
+            dokkaConfiguration {
+                kotlinSourceSet {
+                    samples = setOf("/samples/collections/collections.kt")
+                }
+            }
+
+            sampleFile("/samples/collections/collections.kt", fqPackageName = "samples.collections") {
+                +"""
+                    import samples.*
+                    import kotlin.test.*
+
+                    @RunWith(Enclosed::class)
+                    class Collections {
+                            fun someSample() {
+                                kotlin.test.assertFalse(empty.isNotEmpty())
+                            }
+                        }
+                    }
+                """
+            }
+        }
+        testProject.useServices { context ->
+            val sample = sampleAnalysisEnvironmentCreator.use {
+                resolveSample(
+                    sourceSet = context.singleSourceSet(),
+                    fullyQualifiedLink = "samples.collections.Collections.someSample"
+                )
+            }
+            assertNotNull(sample)
+
+            val expectedImports = listOf(
+                "kotlin.test.*"
+            )
+
+            val expectedBody = "kotlin.test.println(\"empty.isNotEmpty() is \${empty.isNotEmpty()}\") // false"
```

Comment:
The current implementation keeps the old behavior. 
But what do you expect here for fq `kotlin.test.assertFalse`?